### PR TITLE
Docs page on seat type

### PR DESCRIPTION
**Pull request by nstokoe**
We are planning to add seat type labels to the app itself, so it would be nice to have a place to link to explaining what this concept is.

### CODE REVIEW

File: pages/docs/users-and-permissions/seat-types.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+> Editor seat type, and similarly, a role named "Collaborator" can be adjusted to have permissions that qualify it as an Editor seat type.
+
+## Understanding seat types
+
+### Editor Seats
+
+Editors are users who can modify data models and have broader control over the analytics environment.
+A user is considered an Editor if they have any of the following permissions:
+
+- `CREATE_DATA_MODEL`
```
Comment (annanixon) [inline pages/docs/users-and-permissions/seat-types.mdx:18]: Should build permissions be on here as well?

Comment (nstokoe) [reply pages/docs/users-and-permissions/seat-types.mdx:18]: Good question. Ultimately this is a business decision. @trucklos @vndrewlee 

So the proposal would to also include `CREATE_DATA_MODEL_REPO_BUILD` to the list of permissions that qualifies a role/user as an Editor seat

Comment (dse0) [reply pages/docs/users-and-permissions/seat-types.mdx:18]: Correct me if I'm wrong, but it seems to me that even if you have `CREATE_DATA_MODEL_REPO_BUILD`, you still need to have the appropriate create / update model permissions to actually change them? https://github.com/gleannyc/glean/blob/2040369d21f0bf4047aed40c68ee53db68ae157b/glean/services/data_ops/model_conversions.py#L413

Comment (nstokoe) [reply pages/docs/users-and-permissions/seat-types.mdx:18]: @dse0 you are correct, and @luchodev just verified this in the context of the status update permission. So it makes sense that CREATE_DATA_MODEL_REPO_BUILD does not make a role an editor seat-role

**Review by annanixon — APPROVED** [review]
🚀

**Pull request merged**